lib/posix/Kconfig.sched

```
@@ -0,0 +1,10 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config POSIX_SCHED
```

I really ❤️  where this is going. Basically where I was headed too. One minor adjustment, if that's ok, because I'm hoping to try and make POSIX Kconfig option names map directly to POSIX Options / Groups. In the future, this might change _slightly_ though. Can chat a bit about it on Discord if you like.

```suggestion
config POSIX_PRIORITY_SCHEDULING
```